service/controller/resource/app/desired.go

```diff
@@ -145,17 +151,40 @@ func (r *Resource) newApp(cc controllercontext.Context, cr apiv1alpha2.Cluster,
 	}
 }
 
-func (r *Resource) newAppSpecs() []key.AppSpec {
-	switch r.provider {
-	case "aws":
-		return append(key.CommonAppSpecs(), key.AWSAppSpecs()...)
-	case "azure":
-		return append(key.CommonAppSpecs(), key.AzureAppSpecs()...)
-	case "kvm":
-		return append(key.CommonAppSpecs(), key.KVMAppSpecs()...)
-	default:
-		return key.CommonAppSpecs()
+func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]key.AppSpec, error) {
+	req := searcher.Request{
+		ReleaseVersion: key.ReleaseVersion(&cr),
+	}
+
+	res, err := r.clusterClient.Release.Searcher.Search(ctx, req)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var specs []key.AppSpec
+	for _, app := range res.Apps {
```

Comment:
I don't think these code should be encapsulated by another function, WDYT?

Reply:
And so far I didn't think about the possibility, what if `apps` are empty on specific release.

Reply:
Maybe returning an executionFailed error if the apps slice is empty would be a good idea? As there should always be at least one app.

Reply:
@rossf7 I'm curious maybe we would have empty apps on an older release. Maybe it's not the case since we already flattened cluster-operator?

Reply:
Currently this is handled in https://github.com/giantswarm/cluster-operator/blob/3e235cfbc52c6b5fab76e867fe92e71ecae3748d/service/controller/resource/releaseversions/resource.go#L93 so mark as resolved.